tests/kani/SIMD/array_simd_repr.rs

```diff
@@ -0,0 +1,34 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+//! Verify that Kani can properly handle SIMD declared using array syntax.
+
+#![allow(non_camel_case_types)]
+#![feature(repr_simd)]
+
+#[repr(simd)]
+#[derive(PartialEq, Eq, kani::Arbitrary)]
+pub struct i64x2([i64; 2]);
+
+#[kani::proof]
+fn check_diff() {
+    let x = i64x2([1, 2]);
+    let y = i64x2([3, 4]);
+    assert!(x != y);
+}
+
+#[kani::proof]
+fn check_nondet() {
+    let x: i64x2 = kani::any();
+    let y: i64x2 = kani::any();
+    kani::cover!(x != y);
+    kani::cover!(x == y);
```

Comment:
This doesn't enforce a particular result.

Reply:
True, I forgot to rewrite this. I really wish we had an enforce cover. ☺️